code/components/discord/src/DiscordPresence.cpp

```diff
@@ -108,4 +132,41 @@ static InitFunction initFunction([]()
 
 		g_richPresenceChanged = true;
 	});
+
+	fx::ScriptEngine::RegisterNativeHandler("SET_DISCORD_RICH_PRESENCE_ASSET", [](fx::ScriptContext& context)
+	{
+		const char* str = context.GetArgument<const char*>(0);
+
+		if (str)
+		{
+			g_richPresenceOverrideAsset = str;
+		}
+		else
+		{
+			g_richPresenceOverrideAsset = DEFAULT_APP_ASSET;
+		}
+
+		g_richPresenceChanged = true;
+	});
+
+	fx::ScriptEngine::RegisterNativeHandler("SET_DISCORD_APP_ID", [](fx::ScriptContext& context)
+	{
+		const char* str = context.GetArgument<const char*>(0);
+
+		if (str)
+		{
+			g_discordAppId = str;
+		}
+		else
+		{
+			g_discordAppId = DEFAULT_APP_ID;
+		}
+		
+		Discord_Shutdown();
```

Comment:
Can this not be done in the tick routine using a appid != lastappid check (or similar)?